hazelcast/src/hazelcast/client/config.cpp

```diff
@@ -703,12 +704,21 @@ namespace hazelcast {
 
         load_balancer &client_config::get_load_balancer() {
             if (!load_balancer_) {
-                load_balancer_.reset(new impl::RoundRobinLB{});
+                auto index = std::make_shared<std::atomic<size_t>>(0);
+                load_balancer_ = load_balancer().next([=](cluster &c) {
+                    auto members = c.get_members();
+                    if (members.empty()) {
+                        return boost::optional<member>();
+                    }
+                    auto i = index->load();
+                    index->store(i + 1);
```

Comment:
```suggestion
                    auto i = index->fetch_add(1);
```

or

```suggestion
                    auto i = (*index)++;
```

Otherwise two separate threads can get the same index.